sway-lib-std/src/block.sw

```diff
@@ -21,3 +30,51 @@ pub fn timestamp_of_block(block_height: u64) -> u64 {
         timestamp: u64
     }
 }
+
+/// Get the header hash of the block at height `block_height`
+pub fn block_header_hash(block_height: u64) -> Result<b256, BlockHashError> {
+
+    let mut header_hash = ZERO_B256;
+
+    asm(r1: __addr_of(header_hash), r2: block_height) {
+        bhsh r1 r2;
+    };
+
+    // `bhsh` returns b256(0) if the block is not found, so catch this and return an error
+    if header_hash == ZERO_B256 {
+        Result::Err(BlockHashError::BlockHeightTooHigh)
+    } else {
+        Result::Ok(header_hash)
+    }
+}
+
+////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////
+ 
+#[test(should_revert)]
+fn test_block_header_hash_err_current_height() {
```

Comment:
If the condition for `height >= current height` holds then aren't both of these tests doing the same thing despite one passing in a `+1`? I would keep the first test

Reply:
Well, having 2 tests could prevent a regression in the VM; the first tests the `=` condition, and the second tests the `>` condition.